src/Prometheus.Client.MetricServer/MetricServer.HttpListener.cs

```diff
@@ -139,7 +139,7 @@ private void StartListen()
                         response.ContentType = Defaults.ContentType;
                         using (var outputStream = response.OutputStream)
                         {
-                            ScrapeHandler.Process(Registry, outputStream);
+                            ScrapeHandler.ProcessAsync(Registry, outputStream);
```

Comment:
It should await. Either with .GetAwaiter().GetResult() or as a async/await method.

Reply:
Done